pkg/aks/create.go

```diff
@@ -38,219 +38,106 @@ func CreateCluster(ctx context.Context, cred *Credentials, clusterClient *contai
 	}
 
+	virtualNetworkResourceGroup := spec.ResourceGroup
 	if containerservice.NetworkPlugin(to.String(spec.NetworkPlugin)) == containerservice.Azure {
-		networkProfile.NetworkPlugin = containerservice.NetworkPlugin(to.String(spec.NetworkPlugin))
-		virtualNetworkResourceGroup := spec.ResourceGroup
-
 		// If a virtual network resource group is set, use it, otherwise assume it is the same as the cluster
-		if spec.VirtualNetworkResourceGroup != nil {
+		if to.String(spec.VirtualNetworkResourceGroup) != "" {
 			virtualNetworkResourceGroup = to.String(spec.VirtualNetworkResourceGroup)
 		}
 
-		vmNetSubnetID = to.StringPtr(fmt.Sprintf(
-			"/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Network/virtualNetworks/%s/subnets/%s",
-			cred.SubscriptionID,
-			virtualNetworkResourceGroup,
-			to.String(spec.VirtualNetwork),
-			to.String(spec.Subnet),
-		))
-
 		networkProfile.DNSServiceIP = spec.NetworkDNSServiceIP
 		networkProfile.DockerBridgeCidr = spec.NetworkDockerBridgeCIDR
 		networkProfile.ServiceCidr = spec.NetworkServiceCIDR
 		networkProfile.PodCidr = spec.NetworkPodCIDR
 	}
+	managedCluster.ManagedClusterProperties.NetworkProfile = networkProfile
 
-	// Get agent pool profile from config spec
-	agentPoolProfiles := make([]containerservice.ManagedClusterAgentPoolProfile, 0, len(spec.NodePools))
+	agentPoolProfiles := []containerservice.ManagedClusterAgentPoolProfile{}
 	for _, np := range spec.NodePools {
-		if np.OrchestratorVersion == nil {
-			np.OrchestratorVersion = spec.KubernetesVersion
-		}
 		agentProfile := containerservice.ManagedClusterAgentPoolProfile{
-			Name:                np.Name,
-			Count:               np.Count,
-			MaxPods:             np.MaxPods,
-			OsDiskSizeGB:        np.OsDiskSizeGB,
-			OsDiskType:          containerservice.OSDiskType(np.OsDiskType),
-			OsType:              containerservice.OSType(np.OsType),
-			VMSize:              containerservice.VMSizeTypes(np.VMSize),
-			Mode:                containerservice.AgentPoolMode(np.Mode),
-			OrchestratorVersion: np.OrchestratorVersion,
+			Name:         np.Name,
+			Count:        np.Count,
+			MaxPods:      np.MaxPods,
+			OsDiskSizeGB: np.OsDiskSizeGB,
+			OsDiskType:   containerservice.OSDiskType(np.OsDiskType),
+			OsType:       containerservice.OSType(np.OsType),
+			VMSize:       containerservice.VMSizeTypes(np.VMSize),
+			Mode:         containerservice.AgentPoolMode(np.Mode),
+		}
+
+		agentProfile.OrchestratorVersion = spec.KubernetesVersion
+		if to.String(np.OrchestratorVersion) != "" {
+			agentProfile.OrchestratorVersion = np.OrchestratorVersion
 		}
 
 		if np.AvailabilityZones != nil && len(*np.AvailabilityZones) > 0 {
 			agentProfile.AvailabilityZones = np.AvailabilityZones
 		}
 
-		if np.EnableAutoScaling != nil && *np.EnableAutoScaling {
+		if to.StringSlice(np.AvailabilityZones) != nil {
+			agentProfile.AvailabilityZones = np.AvailabilityZones
+		}
+
+		if to.Bool(np.EnableAutoScaling) {
```

Comment:
It has to be like this `if np.EnableAutoScaling != nil && *np.EnableAutoScaling` because of managed fields by Rancher

Reply:
shouldn't `to.Bool()` be the same https://github.com/Azure/go-autorest/blob/main/autorest/to/convert.go#L72 or I'm missing something?